src/q_eol.cc

```diff
@@ -76,35 +75,6 @@ static bool_ quest_eol_gen_hook(void *, void *, void *)
 		done = generate_fracave(y0, x0, xsize, ysize, cutoff, FALSE, TRUE);
 	}
 
-	/* Place a few traps */
-	for (x = xsize - 1; x >= 2; x--)
```

Comment:
Ah, here's your Eol problem. Note that this actually also places Eol himself in the level!